internal/rpmmd/repository.go

```diff
@@ -385,20 +362,22 @@ func runDNF(dnfJsonPath string, command string, arguments interface{}, result in
 }
 
 type rpmmdImpl struct {
-	CacheDir    string
-	RHSM        *RHSMSecrets
-	dnfJsonPath string
+	CacheDir      string
+	subscriptions *rhsm.Subscriptions
+	dnfJsonPath   string
 }
 
 func NewRPMMD(cacheDir, dnfJsonPath string) RPMMD {
+	// Expect that errors in loading are caused by missing subscriptions.
+	subscptions, _ := rhsm.LoadSystemSubscriptions()
```

Comment:
We should at least log the error. Hiding it makes debugging difficult. Also `s/subscptions/subscriptions`. :)

Reply:
We should probably treat the cases of no subscriptions gracefully and not return an error in that case (it is valid to have an unsubscribed system), and then only fail on real errors here.

We should, however, fail (with a helpful error code) in case someone tries to do something later on that requires subscriptions and we have none.

Reply:
I implemented these error handling cases:
 1. The repo file is not there - print a message about it
 2. The repo file is there but it is not possible to open it or parse it - print a different message
 3. User attempts to use rhsm on a system with no subscriptions - suggest that the system must be subscribed